test/Clang/clang_static_library.py

```diff
@@ -32,11 +34,23 @@
 if not test.where_is('clang'):
     test.skip_test("Could not find 'clang', skipping test.\n")
 
+if sys.platform == 'win32':
+    foo_lib = 'foo.lib'
+    archiver = 'mslib'
+    import SCons.Tool.MSCommon as msc
+    if not msc.msvc_exists():
```

Comment:
This test will fail if no mslib right?

Reply:
I assume so. Is that a problem?  As it currently stands, it is failing because there is no 'ar'.

Reply:
Oh, I see.  Possibly: you tell me.  I don't know if there's any "other path" on windows.  I'm not sure how much it makes sense for msvc to have to exist either (as I said, that stanza is copied from another test)... what if people try to install clang for building instead of msvc?  Unfortunately I'm just guessing here...

Reply:
It was my understanding that you need msvc tools to build viable dlls, libs, and exe's with clang? (doesn't it call msvc linker under the hood?)

Reply:
If you use the choco-packaged clang, which is the llvm-built-for-the-Microsoft-way-of-doing-things model of windows. But there are other options too. See for example here:  https://github.com/boostorg/hana/wiki/Setting-up-Clang-on-Windows which sites four possible combinations, two of which helpfully don't appear to work :)

This is why I mentioned use-cases: not sure if the expection is clang will be used both msvc-style and mingw-style.

Reply:
It's probably the case that we need a separate clang-cl tools (-cl aims to process msvc style command line flags instead of "normal/posix" type compiler flags)

Reply:
I saw that... maybe use the github project tab for logging those kinds of work items?

Reply:
might be overkill. just add a tag to the issue maybe. for now until someone asks for it, there's many other issues to work on.